### ISSUE TITLE
Follow up to quick question....

### ISSUE THREAD

**Issue by DasNaughtie** [MassTransit/Sample-Library#3]
Hello again,

Hope all's good.

OK...so here's my sample - apologies if it looks bad as not great with using this editor.

`
[Fact]
public async Task Test()
{
    var date = new DateTime(2023, 1, 1);

    var databaseService = new Mock<IDatabaseService>();
    var transaction = new Mock<IDbContextTransaction>();
    var dateService = new Mock<IDateService>();
    var logger = new Mock<ILogger<CreateClientConsumer>>();

    databaseService
        .Setup(x => x.Client)
        .ReturnsDbSet(new List<Client>());

    databaseService
        .Setup(x => x.Transaction)
        .Returns(transaction.Object);

    dateService
        .Setup(x => x.GetDate())
        .Returns(date);

    await using var provider = new ServiceCollection()
        .AddSingleton(s => databaseService.Object)
        .AddSingleton(s => dateService.Object)
        .AddSingleton(s => logger.Object)
        .AddMassTransitTestHarness(x =>
        {
            x.AddConsumer<CreateClientConsumer>();
            x.UsingInMemory((context, cfg) =>
            {
                cfg.ConfigureEndpoints(context);
            });
        })
        .BuildServiceProvider(true);

    var harness = provider.GetRequiredService<ITestHarness>();

    await harness.Start();

    var client = harness.GetRequestClient<CreateClient>();

    var response = await client.GetResponse<ClientCreated>(new
    {
        TableName = "TableName",
        OrderNumber = "Test",
        Date = date,
        UserId = "SimpsonD"
    });

    var sent = await harness.Sent.Any<ClientCreated>();
    sent.Should().BeTrue();

    var consumed = await harness.Consumed.Any<CreateClient>();
    consumed.Should().BeTrue();
}
`

The test above is designed to test a consumer that writes to DB then publishes a messages to let interested parties know it's complete.  The consumer works fine and I also have a number of integration test that are also working fine...just can't seem to figure out the above - probably something silly.

Thanks in advance.

**Comment by phatboyg** [comment]
That isn't a complete scenario, it's a snippet. You'd need to have a standalone repository that is public to get any input from the community.

**Comment by DasNaughtie** [comment]
My repository is private...finance company so it's not possible but fair enough.  If there's no advice you can offer based on the above then I'll code an example.  Thanks

**Comment by phatboyg** [comment]
You can always work with [support](https://masstransit.io/support) for your organization, if you have a support agreement.

**Comment by DasNaughtie** [comment]
Hi there...hope all's good.

So thought I'd provide a quick update.  Decided I'd see if I could get to the bottom of it (again) after spending a few hours away.  Ended up changing the test design so that it performs a Send instead but was still failing when debugging.  Started to notice that my debugging session was stepping into your code but didn't think anything of it.  Anyway, I accidently click on Run instead of Debug and the test passed!  Debugged it again and it failed.  Weird?!  Then I figured I'd turn off the feature that lets me step into 3rd party code (options -> debugging -> Just My Code.  I use Visual Studio so this has to be turned on) and it worked - now consistently passing.  Probably one of the most unique set of circumstances you've heard in a while!
Anyway, got there in the end.
